include/zephyr/drivers/charger.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CHARGER_H_
+#define ZEPHYR_INCLUDE_DRIVERS_CHARGER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/device.h>
+
+/* Keep these alphabetized wrt property name */
+
+/** Runtime Dynamic Battery Parameters */
+
+/** Indicates if VBUS is present for the charger */
+#define CHARGER_ONLINE	0
+/** Reports whether or not a battery is present */
+#define CHARGER_PRESENT CHARGER_ONLINE + 1
+/** Represents the charging status of the charger */
+#define CHARGER_STATUS	CHARGER_PRESENT + 1
+
+/** Reserved to demark end of common charger properties */
+#define CHARGER_COMMON_COUNT CHARGER_STATUS + 1
+/**
+ * Reserved to demark downstream custom properties - use this value as the actual value may
+ * change over future versions of this API
+ */
+#define CHARGER_CUSTOM_BEGIN CHARGER_COMMON_COUNT + 1
+
+/** Reserved to demark end of valid enum properties */
+#define CHARGER_PROP_MAX UINT16_MAX
```

Comment:
please introduce a new Doxygen group, and group definitions together, see e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/regulator.h#L39

Reply:
ack, i think i did this right